src/Symfony/Component/Cache/Adapter/ProxyAdapter.php

```diff
@@ -138,12 +143,12 @@ private function doSave(CacheItemInterface $item, $method)
             return false;
         }
         $item = (array) $item;
-        $expiry = $item[CacheItem::CAST_PREFIX.'expiry'];
-        $poolItem = $this->pool->getItem($this->namespace.$item[CacheItem::CAST_PREFIX.'key']);
```

Comment:
This is where the double is: `getItem()` is used once to get the item, then here again just to get a proper item to save in the proxyfied pool. But by saving the proxyfied item in the CacheItem itself, we work around this extra fetch (which can be coslty depending on the pool).